packages/material-ui/src/Chip/Chip.js

```diff
@@ -300,12 +308,7 @@ const Chip = React.forwardRef(function Chip(props, ref) {
       onKeyDown(event);
     }
 
-    // Ignore events from children of `Chip`.
-    if (event.currentTarget !== event.target) {
```

Comment:
What do you think of #10324?

Reply:
Yeah let me clean this up and then add missing tests to see why do or do not need this.

Reply:
Feel free to ignore my comment, it's only a draft after all :)

Reply:
So yeah the issue is about being consistent. The DOM already has the functionality to prevent this behavior with stopPropagation or preventDefault. I'll reintroduce this just to prevent potential breaking changes. But we should not add more of this kind of behavior since there are existing mechanics to resolve these issues.

Reply:
The keydown code is very weird at the moment. I'll leave it as is but the input inside chip case should be handled with preventDefault.

Reply:
Seems like this was either added before this was handled accordingly in ButtonBase or when tests were shallow.